homeassistant/components/launch_library/sensor.py

```diff
@@ -190,31 +204,33 @@ def __init__(
     @property
     def native_value(self) -> datetime | str | int | None:
         """Return the state of the sensor."""
-        if self._next_launch is None:
+        if self._next_event is None:
             return None
-        return self.entity_description.value_fn(self._next_launch)
+        return self.entity_description.value_fn(self._next_event)
 
     @property
     def extra_state_attributes(self) -> dict[str, Any] | None:
         """Return the attributes of the sensor."""
-        if self._next_launch is None:
+        if self._next_event is None:
             return None
-        return self.entity_description.attributes_fn(self._next_launch)
+        return self.entity_description.attributes_fn(self._next_event)
 
     @property
     def available(self) -> bool:
         """Return if the sensor is available."""
-        return super().available and self._next_launch is not None
+        return super().available and self._next_event is not None
 
     @callback
     def _handle_coordinator_update(self) -> None:
         """Handle updated data from the coordinator."""
         if self.entity_description.key == "starship_launch":
-            launches = self.coordinator.data["starship_events"].upcoming.launches
+            events = self.coordinator.data["starship_events"].upcoming.launches
+        elif self.entity_description.key == "starship_event":
+            events = self.coordinator.data["starship_events"].upcoming.events
```

Comment:
please add this to the diagnostics platform as well :+1:

Reply:
Doh! 🤦 I can't believe i forgot this 😆